File: app/src/main/java/fr/free/nrw/commons/wikidata/WikidataEditListener.java
package fr.free.nrw.commons.wikidata;

public abstract class WikidataEditListener {

  protected WikidataP18EditListener wikidataP18EditListener;

  public abstract void onSuccessfulWikidataEdit();

  public void setAuthenticationStateListener(WikidataP18EditListener wikidataP18EditListener) {
    this.wikidataP18EditListener = wikidataP18EditListener;
  }

  public interface WikidataP18EditListener {

    void onWikidataEditSuccessful();
  }
}

File: app/src/main/java/fr/free/nrw/commons/wikidata/WikidataEditListener.kt
package fr.free.nrw.commons.wikidata

abstract class WikidataEditListener {
    var authenticationStateListener: WikidataP18EditListener? = null

    abstract fun onSuccessfulWikidataEdit()

    interface WikidataP18EditListener {
        fun onWikidataEditSuccessful()
    }
}
